### ISSUE TITLE
JavaScript component generation for Vue

### ISSUE THREAD

**Issue by yamachu** [dotnet/aspnetcore#40539]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

I excited when I read below article.
https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-net-6-rc-1/#generate-angular-and-react-components-using-blazor

I wanted to develop modern frontend applications with a more strictly typed language (such as C# ) for logic.
I had a good impression of it when I actually used it because it fits that usecase.

When I demonstrated with the community and developers I knew, I was able to generate interest.
However, I also received the following comments...

> How is Vue?

At this time, there are no sample implementations in Vue in the sample repository.
https://github.com/aspnet/samples/tree/main/samples/aspnetcore/blazor/JSComponentGeneration

It is possible to implement it if users understand the content, but it is a little difficult to recommend it to people who are interested.

### Describe the solution you'd like

Provide Vue samples.

I have my own samples for developers, but I would prefer to have them provided in the official sample repository.
https://github.com/yamachu/aspnet-samples/compare/main...add-vue-sample

### Additional context

Feedback for Generate Angular and React components using Blazor

I use React and Blazor interoperation not for reusing Blazor's UI, but for providing logic to the React side.

By using the `DotnetReference` object, we were able to use the methods of the Blazor component without polluting the global space.
Therefore, we were able to use the Blazor component as a service class component.

samples:
https://github.com/yamachu/BlazorWithReactSample/blob/3678154d3cb9264d2c7656209439b99417434a9f/React/src/blazor/Counter.ts
https://github.com/yamachu/BlazorWithReactSample/blob/3678154d3cb9264d2c7656209439b99417434a9f/BlazorWebAssembly/Pages/Counter.razor

Thanks to the interoperation between Blazor and JavaScript, it was easier to use Blazor components, including Native-dependent wasm.
In other words, memory management, etc., which are troublesome in JavaScript, could be implemented on the C# side and provided to the React component.

I am making good use of it in my project, thank you.

**Comment by javiercn** [comment]
@yamachu thanks for contacting us.

We provided a sample on how to do this for React and Angular. We don't have plans to provide guidance/samples on how to do this for other JavaScript frameworks as there are many and the pattern to follow is similar and can be implemented by people interested if needed.
